components/Slider.js

```diff
@@ -0,0 +1,260 @@
+    const valuePercentage = (((value - min) / range) * 100);
+
+    let left;
+    let newValue;
+    left = valuePercentage;
+    newValue = value;
+
+    if (evt) {
+      const { type } = evt;
+
+      if (type === 'keydown') {
+        const direction = {
+          40: -1, // decreasing
+          37: -1, // decreasing
+          38: 1, // increasing
+          39: 1, // increasing
+        }[evt.which];
+
+        if (direction !== undefined) {
+          const multiplier = evt.shiftKey === true
+            ? (range / step) / stepMuliplier
+            : 1;
+          const stepMultiplied = step * multiplier;
+          const stepSize = (stepMultiplied / range) * 100;
+          left = valuePercentage + (stepSize * direction);
+          newValue = Number(value) + (stepMultiplied * direction);
+        }
+      }
+      if (type === 'mousemove' || type === 'click' || type === 'touchmove') {
+        const clientX = evt.touches ? evt.touches[0].clientX : evt.clientX;
+        const track = this.track.getBoundingClientRect();
+        const unrounded = ((clientX - track.left) / track.width);
+        const rounded = Math.round(((range * unrounded) / step)) * step;
+        left = (((rounded - min) / range) * 100);
+        newValue = rounded;
+      }
+    }
+
+    if (newValue <= Number(min)) {
+      left = 0;
+      newValue = min;
+    }
+    if (newValue >= Number(max)) {
+      left = 100;
+      newValue = max;
+    }
+
+    return { left, newValue };
+  }
+
+  handleMouseStart = () => {
+    this.element.ownerDocument.addEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.addEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleMouseEnd = () => {
+    this.element.ownerDocument.removeEventListener('mousemove', this.updatePosition)
+    this.element.ownerDocument.removeEventListener('mouseup', this.handleMouseEnd)
+  }
+
+  handleTouchStart = () => {
+    this.element.ownerDocument.addEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.addEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.addEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  handleTouchEnd = () => {
+    this.element.ownerDocument.removeEventListener('touchmove', this.updatePosition);
+    this.element.ownerDocument.removeEventListener('touchup', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchend', this.handleTouchEnd);
+    this.element.ownerDocument.removeEventListener('touchcancel', this.handleTouchEnd);
+  }
+
+  handleChange = (evt) => {
+    this.setState({ value: evt.target.value });
+    this.updatePosition();
+  }
+
+  render() {
+    const {
+      className,
+      hideTextInput,
+      id,
+      onChange,
+      min,
+      minLabel,
+      max,
+      maxLabel,
+      labelText, 
+      step,
+      required,
+      disabled,
```

Comment:
TL;DR: add `...other` here as well, see [original comment for details](https://github.com/carbon-design-system/carbon-components-react/pull/65#issuecomment-317458907)